### PR TITLE
zimfw: init at 1.11.0

### PR DESCRIPTION
**Pull request by joedevivo**
###### Description of changes

[zimfw](https://zimfw.sh) is a framework for zsh that is very lightweight and compiles zsh scripts for speed at runtime.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6730,6 +6730,12 @@
     githubId = 3967312;
     name = "Jocelyn Thode";
   };
+  joedevivo = {
+    email = "55951+joedevivo@users.noreply.github.com";
+    github = "joedevivo";
+    githubId = 55951;
+    name = "Joe DeVivo";
+  };
```
Comment (lilyinstarlight) [inline maintainers/maintainer-list.nix:6738]: This is buried deep within the manual (and we should maybe add it to CONTRIBUTING.md too...), but additions to `maintainers/maintainter-list.nix` are supposed to be in their own commits titled "maintainers: add [name]": https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

File: pkgs/shells/zsh/zimfw/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zimfw";
+  version = "v1.11.0";
+  src = fetchFromGitHub {
+    owner = "zimfw";
+    repo = pname;
+    rev = version;
```
Comment (lilyinstarlight) [inline pkgs/shells/zsh/zimfw/default.nix:9]:
```suggestion
  pname = "zimfw";
  version = "1.11.0";
  src = fetchFromGitHub {
    owner = "zimfw";
    repo = "zimfw";
    rev = "v${version}";
```

Using `repo = pname` is now discouraged afaik and the version should be just the version number without the prefix

File: pkgs/shells/zsh/zimfw/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ## zim only needs this one file to be installed.
+    sparseCheckout = [ "zimfw.zsh" ];
+    sha256 = "sha256-BmzYAgP5Z77VqcpAB49cQLNuvQX1qcKmAh9BuXsy2pA=";
+  };
+  strictDeps = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontPatch = true;
```
Comment (lilyinstarlight) [inline pkgs/shells/zsh/zimfw/default.nix:17]:
```suggestion
```

The other two are fine, but is there any compelling reason to set `dontPatch = true`? (especially if, e.g., someone overrides to add a patch file)

File: pkgs/shells/zsh/zimfw/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r $src/zimfw.zsh $out/
+  '';
```
Comment (lilyinstarlight) [inline pkgs/shells/zsh/zimfw/default.nix:22]:
```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out
    cp -r $src/zimfw.zsh $out/
    
    runHook postInstall
  '';
```

When overriding phases like this, remember to still run the pre/post hooks

**Review by lilyinstarlight — COMMENTED** [review]
This looks pretty good, I just left a handful of nits really

File: pkgs/shells/zsh/zimfw/default.nix (hunk 3 above)
Comment (joedevivo) [reply pkgs/shells/zsh/zimfw/default.nix:17]: tbh I based everything on `../zplug/default.nix`. I'm happy to apply all of your suggestions.

Comment (lilyinstarlight) [reply pkgs/shells/zsh/zimfw/default.nix:17]: I think the `repo = pname` one isn't that bad if you really want to leave it, but all of the other changes probably should be applied (and zplug was added a few years ago so some best practices have been changed or more strictly enforced)

**Comment by joedevivo** [comment]
@lilyinstarlight I've made the changes. Ready for review.

**Review by lilyinstarlight — APPROVED** [review]
Diff looks good to me and it builds successfully

Result of `nixpkgs-review pr 208919` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zimfw</li>
  </ul>
</details>

**Comment by joedevivo** [comment]
Thanks @lilyinstarlight, can you merge it?

**Comment by lilyinstarlight** [comment]
> Thanks @lilyinstarlight, can you merge it?

Nope, I don't have write access, sorry. If no one merged it in the next day or so, I was gonna pester someone in that discourse thread for already-reviewed PRs to merge it
